curl: Argument List Too Long

Saw this error while trying to use this action. 

```
2019-11-20T18:47:31.3161806Z Writing {manual edit: filename} (503268 bytes)
2019-11-20T18:47:31.3167681Z Skipped writing 2 intermediate files (use --keep-intermediates to keep them)
2019-11-20T18:47:31.3530256Z   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2019-11-20T18:47:31.3530620Z                                  Dload  Upload   Total   Spent    Left  Speed
2019-11-20T18:47:31.3530687Z 
2019-11-20T18:47:31.5159769Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-20T18:47:31.5160199Z 100   116  100   116    0     0    704      0 --:--:-- --:--:-- --:--:--   703
2019-11-20T18:47:31.5160534Z 100   116  100   116    0     0    704      0 --:--:-- --:--:-- --:--:--   703
2019-11-20T18:47:31.5200413Z /entrypoint.sh: 10: /entrypoint.sh: curl: Argument list too long
2019-11-20T18:47:32.1299451Z ##[error]Docker run failed with exit code 2
```

I think https://github.com/vinay0410/tectonic-action/blob/master/entrypoint.sh#L16 is the line at issue. [StackExchange](https://unix.stackexchange.com/questions/174350/curl-argument-list-too-long) says "if the base64-encoded file is too big to fit in the argument list you are going to have to pass it via a file."

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -7,7 +7,7 @@ tectonic $1
 
 OUTPUT_PDF="${1%.*}.pdf"
 
-STATUSCODE=$(curl --silent --output /dev/stderr --write-out "%{http_code}" -i -X PUT -H "Authorization: token $GITHUB_TOKEN" -d '{
+echo '{
   "message": "'"update $OUTPUT_PDF"'",
   "committer": {
     "name": "Tectonic Action",
@@ -15,7 +15,11 @@ STATUSCODE=$(curl --silent --output /dev/stderr --write-out "%{http_code}" -i -X
   },
   "content": "'"$(base64 -w 0 $OUTPUT_PDF)"'",
   "sha": '$(curl -X GET https://api.github.com/repos/vinay0410/webpage/contents/$OUTPUT_PDF | jq .sha)'
-}' https://api.github.com/repos/vinay0410/webpage/contents/$OUTPUT_PDF)
+}' >> payload.json
+
+STATUSCODE=$(curl --silent --output /dev/stderr --write-out "%{http_code}" \
+            -i -X PUT -H "Authorization: token $GITHUB_TOKEN" -d @payload.json \
+            https://api.github.com/repos/vinay0410/webpage/contents/$OUTPUT_PDF)
 
 if [ $((STATUSCODE/100)) -ne 2 ]; then
   echo "Github's API returned $STATUSCODE"

